src/renderer/views/LobbyView/index.tsx

```diff
@@ -1,26 +1,35 @@
 import { observer } from "mobx-react";
 import React, { useEffect } from "react";
-import { useHistory } from "react-router";
+import { useHistory, useLocation } from "react-router";
 import Layout from "src/renderer/components/core/Layout";
-import { useActivationStatus } from "src/utils/useActivationStatus";
+import { useCheckContract } from "src/utils/useCheckContract";
 import { useStore } from "src/utils/useStore";
 
 function LobbyView() {
   const { account, game } = useStore();
-  const { loading, activated, error } = useActivationStatus();
+  const { loading, error, approved, requested, stopPolling } =
+    useCheckContract(true);
   const history = useHistory();
+  const { search } = useLocation();
 
   useEffect(() => {
-    if (loading || activated || account.activationKey) return;
-    history.push("/register/activationKey");
-  }, [history, loading, activated, account.activationKey]);
+    if (loading || approved || search !== "") return;
+    if (!requested) {
+      stopPolling();
+      history.push("/register/getPatron");
+    } else {
+      stopPolling();
+      history.push("/register/pledgeWait");
+    }
+  }, [history, loading, approved, requested]);
```

Comment:
```suggestion
  }, [history, loading, approved, requested, search]);
```

or, was it intended?